### PR TITLE
Get base info

### PR DESCRIPTION
**Pull request by Tomasvrba**
This adds backend rpc call to the app to get base info implemented on the base here: https://github.com/digitalbitbox/bitbox-base/pull/218

I did this pretty quickly to test the middleware functionality so it might need some tweaks

### CODE REVIEW

**Comment by 0xB10C** [comment]
https://github.com/digitalbitbox/bitbox-wallet-app/blob/daf709781392c070a6e3edf5676f207d8b080910/backend/bitboxbase/rpcclient/websocket.go#L45

Needs a bump, since the GetBaseInfo RPC response is bigger. Long ErrorResponses (512+ bytes) don't get send as well. So I'd say at least 4096 or more. Should think about a max ErrorResponse.Message size as well...

File: backend/bitboxbase/handlers/handlers.go [hunk 1]
```diff
@@ -90,6 +91,7 @@ func NewHandlers(
 	handleFunc("/set-root-password", handlers.postSetRootPasswordHandler).Methods("POST")
 	handleFunc("/shutdown-base", handlers.postShutdownBaseHandler).Methods("POST")
 	handleFunc("/reboot-base", handlers.postRebootBaseHandler).Methods("POST")
+	handleFunc("/base-info", handlers.getBaseInfoHandler).Methods("GET")
```
Comment (0xB10C) [inline backend/bitboxbase/handlers/handlers.go]: The other `GET`s are defined at the top. Maybe move this there for consistency.

**Comment by Tomasvrba** [comment]
> Needs a bump, since the GetBaseInfo RPC response is bigger. Long ErrorResponses (512+ bytes) don't get send as well. So I'd say at least 4096 or more. Should think about a max ErrorResponse.Message length as well.

Bumped to 16384 as discussed with 0xB10C

**Comment by Tomasvrba** [comment]
> Looks good!
> 
> Should we define JSON names for the values of the GetBaseInfoResponse struct? The
> JSON marshaling currently uses the uppercase variable names:
> 
> ```json
> {
>   "baseInfo": {
>     "ErrorResponse": {
>       "Success": true,
>       "Code": "",
>       "Message": ""
>     },
>     "Status": "-PLACEHOLDER-",
>     "Hostname": "bitbox-base",
>     "MiddlewareLocalIP": "",
>     "MiddlewareLocalPort": "8845",
>     "MiddlewareTorOnion": "blablalba.onion",
>     "MiddlewareTorPort": "-PLACEHOLDER-",
>     "IsTorEnabled": true,
>     "IsBitcoindListening": true,
>     "FreeDiskspace": 148830826496,
>     "TotalDiskspace": 491182030848,
>     "BaseVersion": "0.0.3",
>     "BitcoindVersion": "0.18.1",
>     "LightningdVersion": "0.7.2.1",
>     "ElectrsVersion": "0.7.0"
>   },
>   "success": true
> }
> ```

Yeah, good idea. I will add that today

File: backend/bitboxbase/rpcclient/rpcclient.go [hunk 2]
```diff
@@ -415,3 +415,14 @@ func (rpcClient *RPCClient) RebootBase() (rpcmessages.ErrorResponse, error) {
 	}
 	return reply, nil
 }
+
+// GetBaseInfo makes a synchronous rpc call to the base and returns the SampleInfoResponse struct
```
Comment (0xB10C) [inline backend/bitboxbase/rpcclient/rpcclient.go]: Comment is wrong here. It's a `GetBaseInfoResponse` that's being returned.

**Pull request merged**